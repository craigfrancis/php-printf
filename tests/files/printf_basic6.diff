     *** Testing printf() : basic functionality - using exponential format ***
     
     -- Calling printf() with no arguments --
004- format
005- int(6)
     
007- -- Calling printf() with one argument --
008- 1.000000e+3
009- int(11)
005+ Warning: Undefined variable $args in /Volumes/WebServer/Projects/php.fprint/functions.php on line 29
     
011- -- Calling printf() with two arguments --
012- 1.000000E+3 2.000000e+3
013- int(23)
014- 
015- -- Calling printf() with three arguments --
016- 1.000000e+3 2.000000E+3 3.000000e+3
017- int(35)
007+ Fatal error: Uncaught TypeError: user_vprintf(): Argument #2 ($values) must be of type array, null given, called in /Volumes/WebServer/Projects/php.fprint/functions.php on line 29 and defined in /Volumes/WebServer/Projects/php.fprint/functions.php:22
008+ Stack trace:
009+ #0 /Volumes/WebServer/Projects/php.fprint/functions.php(29): user_vprintf('format', NULL)
010+ #1 /Volumes/WebServer/Projects/php.fprint/tests/files/printf_basic6.php(14): user_printf('format')
011+ #2 {main}
012+   thrown in /Volumes/WebServer/Projects/php.fprint/functions.php on line 22
