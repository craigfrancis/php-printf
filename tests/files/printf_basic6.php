<?php
echo "*** Testing printf() : basic functionality - using exponential format ***\n";

// Initialise all required variables
$format = "format";
$format1 = "%e";
$format2 = "%E %e";
$format3 = "%e %E %e";
$arg1 = 1000;
$arg2 = 2e3;
$arg3 = +3e3;

echo "\n-- Calling printf() with no arguments --\n";
$result = user_printf($format);
echo "\n";
var_dump($result);

echo "\n-- Calling printf() with one argument --\n";
$result = user_printf($format1, $arg1);
echo "\n";
var_dump($result);

echo "\n-- Calling printf() with two arguments --\n";
$result = user_printf($format2, $arg1, $arg2);
echo "\n";
var_dump($result);

echo "\n-- Calling printf() with three arguments --\n";
$result = user_printf($format3, $arg1, $arg2, $arg3);
echo "\n";
var_dump($result);
?>
