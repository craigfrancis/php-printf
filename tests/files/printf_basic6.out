*** Testing printf() : basic functionality - using exponential format ***

-- Calling printf() with no arguments --

Warning: Undefined variable $args in /Volumes/WebServer/Projects/php.fprint/functions.php on line 29

Fatal error: Uncaught TypeError: user_vprintf(): Argument #2 ($values) must be of type array, null given, called in /Volumes/WebServer/Projects/php.fprint/functions.php on line 29 and defined in /Volumes/WebServer/Projects/php.fprint/functions.php:22
Stack trace:
#0 /Volumes/WebServer/Projects/php.fprint/functions.php(29): user_vprintf('format', NULL)
#1 /Volumes/WebServer/Projects/php.fprint/tests/files/printf_basic6.php(14): user_printf('format')
#2 {main}
  thrown in /Volumes/WebServer/Projects/php.fprint/functions.php on line 22